instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/propagators/CompositeTextMapPropagatorFactory.java

```diff
@@ -0,0 +1,88 @@
+import io.opentelemetry.api.trace.propagation.W3CTraceContextPropagator;
+import io.opentelemetry.context.propagation.TextMapPropagator;
+import io.opentelemetry.extension.aws.AwsXrayPropagator;
+import io.opentelemetry.extension.trace.propagation.B3Propagator;
+import io.opentelemetry.extension.trace.propagation.JaegerPropagator;
+import io.opentelemetry.extension.trace.propagation.OtTracePropagator;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import org.springframework.beans.factory.BeanFactory;
+import org.springframework.util.ClassUtils;
+
+/** Factory of composite {@link TextMapPropagator}. Defaults to W3C and BAGGAGE. */
+public final class CompositeTextMapPropagatorFactory {
+
+  static TextMapPropagator getCompositeTextMapPropagator(
+      BeanFactory beanFactory, List<PropagationType> types) {
+
+    Set<TextMapPropagator> propagators = new HashSet<>();
+
+    for (PropagationType type : types) {
+      switch (type) {
+        case b3:
+          if (isOnClasspath("io.opentelemetry.extension.trace.propagation.B3Propagator")) {
+            propagators.add(
+                beanFactory
+                    .getBeanProvider(B3Propagator.class)
+                    .getIfAvailable(B3Propagator::injectingSingleHeader));
+          }
+          break;
+        case b3multi:
+          if (isOnClasspath("io.opentelemetry.extension.trace.propagation.B3Propagator")) {
+            propagators.add(
+                beanFactory
+                    .getBeanProvider(B3Propagator.class)
+                    .getIfAvailable(B3Propagator::injectingMultiHeaders));
+          }
+          break;
+        case jaeger:
+          if (isOnClasspath("io.opentelemetry.extension.trace.propagation.JaegerPropagator")) {
+            propagators.add(
+                beanFactory
+                    .getBeanProvider(JaegerPropagator.class)
+                    .getIfAvailable(JaegerPropagator::getInstance));
+          }
+          break;
+        case ottrace:
+          if (isOnClasspath("io.opentelemetry.extension.trace.propagation.OtTracerPropagator")) {
+            propagators.add(
+                beanFactory
+                    .getBeanProvider(OtTracePropagator.class)
+                    .getIfAvailable(OtTracePropagator::getInstance));
+          }
+          break;
+        case xray:
+          if (isOnClasspath("io.opentelemetry.extension.aws.AwsXrayPropagator")) {
+            propagators.add(
+                beanFactory
+                    .getBeanProvider(AwsXrayPropagator.class)
+                    .getIfAvailable(AwsXrayPropagator::getInstance));
+          }
+          break;
+        case tracecontext:
+          propagators.add(W3CTraceContextPropagator.getInstance());
+          break;
+        case baggage:
+          propagators.add(W3CBaggagePropagator.getInstance());
+          break;
+      }
+
+      propagators.add(TextMapPropagator.noop());
```

Comment:
Is this needed?

Reply:
In case user supplies a propagator that we don't support (or simply a wrong one). I guess now that I'll be converting enums to strings it would make sense to move that line to a default case in the switch, correct me if I'm wrong.

Reply:
That should be fine even without that - if you look at `TextMapPropagator#composite()` implementation you'll see that it handles an empty list of propagators correctly. I think it's fine to rely on that.

Reply:
Right, it does the `isEmpty` check and then supplies noop if needed. Thanks!